packages/blockchain-wallet-v4/src/redux/data/activity/selectors.ts

```diff
@@ -0,0 +1,44 @@
+import moment from 'moment'
+
+import {
+  InterestTransactionType,
+  RemoteDataType,
+  SBOrderType,
+  SBTransactionType
+} from 'core/types'
+
+import { liftN } from 'ramda'
+
+import { NabuProducts, NabuTxType, SUCCESS_STATUS } from './types'
+import { RootState } from 'data/rootReducer'
+
+export const getActivity = (state: RootState) => state.dataPath.activity
+
+export const getCustodialActivityStatus = (
+  state: RootState
+): RemoteDataType<string, typeof SUCCESS_STATUS> => {
+  const statuses: Array<RemoteDataType<string, typeof SUCCESS_STATUS>> = []
+  for (const product of NabuProducts) {
+    for (const type of NabuTxType) {
+      statuses.push(state.dataPath.activity[product][type].status)
+    }
+  }
+
+  return liftN(
+    NabuProducts.length * NabuTxType.length,
+    (...args) => args
+  )(...statuses)
+}
```

Comment:
I like this; to be honest it took me a bit to get how it works

Reply:
What do you like about it?

Reply:
isolation/separation of statuses